src/OrchardCore.Modules/OrchardCore.Contents/Controllers/AdminController.cs

```diff
@@ -434,7 +434,7 @@ public Task<IActionResult> EditPOST(string contentItemId, [Bind(Prefix = "submit
         {            
             if (submitSave == "submit.SaveAndContinue")
             {
-                returnUrl = Request.Path.Add(Request.QueryString);
+                returnUrl = Request.PathBase + Request.Path.Add(Request.QueryString);
```

Comment:
Could we change this so that when returnUrl is empty it stays on the same page?
I'd like to prevent the usage of `PathBase` as much as possible.

Reply:
I added this code after your review. You wanted to remove some code that does not seemed necessary. So you suggested adding the current path as a returnUrl. But I did not realized that it could cause this issue on tenants, and that it would need resorting to PathBase.
Not sure how to fix that. The only way I found is to return to the code as it was before the review.
It is working nicely on tenants (I tested it) and It does not need PathBase.  And its does not seems to be more code after all.
What do you think?

Reply:
This is the link to the review in case you want to take a look:
https://github.com/OrchardCMS/OrchardCore/pull/2133/files/9d9d0585ab6b18fd62881bab341c5a2df4aec3df#r205573080